docs/2.10.0/docs/canton/usermanual/packagemanagement.rst

```diff
@@ -61,235 +61,183 @@ Understanding Package Vetting
-See :ref:`topology.vetted_packages.authorize <topology.vetted_packages.authorize>` for more detail.
+.. snippet:: package_dar_vetting
+    .. success:: participant1.dars.vetting.disable(examplesDependencyDarHash)
 
+then we can disable the vetting for the "CantonExamples" DAR as well.
 
-Removing Packages
-~~~~~~~~~~~~~~~~~
+.. snippet:: package_dar_vetting
+    .. success:: participant1.dars.vetting.disable(darHash)
 
-Canton also supports removing individual packages, giving the user more fine-grained control over the system.
-Packages can be removed if the package satisfies the following two requirements:
+Advanced vetting concepts
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
-- The package must be unused. This means that there shouldn't be an active contract corresponding to the package.
+.. note::
+    This section concentrates on lower-level details of package topology state and commands.
+    For most use-cases, the high-level vetting APIs mentioned above should be sufficient.
+    Usage of the lower-level topology APIs can lead to inconsistencies in the participant's topology state
+    and is advised only by expert users.
 
-- The package must not be vetted. This means there shouldn't be an active vetting transaction corresponding to the package.
+Package topology states
+^^^^^^^^^^^^^^^^^^^^^^^
 
-The following tutorial shows how to remove a package using the Canton console. The first step is to upload and identify the
-package ID for the package to be removed.
+With respect to a participant, a package can be in one of the following states:
 
-.. snippet:: package_dar_removal
-    .. success:: val darHash = participant1.dars.upload("dars/CantonExamples.dar")
-    .. success:: val packageId = participant1.packages.find("Iou").head.packageId
+- **Not found** on the participant: The package does not exist in the local participant stores and it can't be referenced in any request to the participant node.
 
-Package removal will initially fail as, by default, uploading the DAR will add a vetting transaction for the package:
+- **Unknown**: The package may exist in the local participant stores, but it has no associated topology transaction issued by the participant node (i.e. it is unknown topology-wise). A package pertaining to a DAR that was uploaded with the vetting flag disabled is unknown.
 
-.. snippet:: package_dar_removal
-    .. failure:: participant1.packages.remove(packageId)
+- **Check-only**: The package appears at least in a `CheckOnlyPackages` topology transaction and it allows a participant to announce that a collection of Daml packages are known but can only be used for validating pre-existing contracts on the ledger and not for executing new Daml transactions. This concept has been introduced in protocol version 7 for supporting :ref:`Smart contract upgrades <smart-contract-upgrades>`.
```

Comment:
```suggestion
- **Check-only**: The package appears in a `CheckOnlyPackages` topology transaction and it allows a participant to announce that a collection of Daml packages is known, but it can only be used to validate preexisting contracts on the ledger, not for executing new Daml transactions. This concept has been introduced in protocol version 7 to support :ref:`Smart contract upgrades <smart-contract-upgrades>`.
```